### PR TITLE
Added registration page

### PR DESCRIPTION
**Pull request by chen-robert**
Added a lot of frameworking and a sample registration page. I think we should merge this now before the branch gets too bloated.

### CODE REVIEW

File: client/src/components/registration.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export default withStyles({
+  root: {
+    padding: '25px'
+  },
+  submit: {
+    'margin-top': '25px'
+  }
+}, class Register extends Component {
+  componentDidMount () {
+    document.title = 'Registration'
```
Comment (ginkoid) [inline client/src/components/registration.js]: Should we append the CTF name to the title? Maybe `Registration | redpwnCTF 2020`.

File: client/src/components/registration.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Component } from 'preact'
+import withStyles from './jss'
+
+export default withStyles({
+  root: {
+    padding: '25px'
+  },
+  submit: {
+    'margin-top': '25px'
```
Comment (ginkoid) [inline client/src/components/registration.js]: This can be `marginTop` because of `jss-camel-case`.

File: client/src/api/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: 'Bearer ' + localStorage.getItem('token')
+    },
+    body: JSON.stringify(data)
+  })
+    .json()
+    .catch(err => {
+      console.debug(err)
+      badToken()
```
Comment (ginkoid) [inline client/src/api/index.js]: Network errors are not necessarily a sign that the token is invalid.

File: client/src/api/auth.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { request } from './index'
+
+export const login = ({ teamToken }) => {
+  return request('POST', '/auth/verify', {
+    teamToken
+  })
+}
+
+export const signup = ({ email, name, division, register }) => {
+  return request('POST', '/auth/signup', {
```
Comment (ginkoid) [inline client/src/api/auth.js]: `/auth/signup` isn't a thing.

File: client/src/components/registration.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        <div class={classes.root + ' col-6'}>
+          <div class='form-section'>
+            <div class='input-control'>
+              <input class='input-contains-icon' id='name' name='name' placeholder='Team Name' type='text' value='' />
+              <span class='icon'>
+                <i class='fa fa-wrapper fa-user-circle small' />
+              </span>
+            </div>
+            <div class='input-control'>
+              <input class='input-contains-icon' id='email' name='email' placeholder='Email' type='text' value='' />
```
Comment (ginkoid) [inline client/src/components/registration.js]: Input box should probably be their own components.

File: client/src/api/index.js (hunk 3 above)
Comment (chen-robert) [reply client/src/api/index.js]: Will fix in separate PR

File: client/src/components/registration.js (hunk 1 above)
Comment (chen-robert) [reply client/src/components/registration.js]: Added to configuration file in `client/config`

File: client/src/components/registration.js (hunk 5 above)
Comment (chen-robert) [reply client/src/components/registration.js]: Will refactor in separate PR

**Pull request merged**